### PR TITLE
Reduce 4.14  (new formula) 

### PR DESCRIPTION
**Pull request by eunos-1128**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source --verbose --keep-tmp ./Formula/<FORMULA>.rb`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your formula have no offenses with `brew style /path/to/formula.rb`?
- [ ] Does your formula pass `brew audit --formula brewsci/bio/<FORMULA> --online --git --skip-style`?
- [ ] Does your formula pass `brew linkage --cached --test --strict brewsci/bio/<FORMULA>` after manual installation?

-----

### CODE REVIEW

File: Formula/reduce.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Reduce < Formula
+  desc "Tool for adding and correcting hydrogens in PDB files"
+  homepage "https://github.com/rlabduke/reduce"
+  url "https://github.com/rlabduke/reduce/archive/refs/tags/v4.14.tar.gz"
+  version "4.14"
```
Comment (YoshitakaMo) [inline Formula/reduce.rb:5]:
```suggestion
```
`version` info is not required when the version name can be scanned from the filename.

File: Formula/reduce.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  sha256 "62e61cce221fff76b5834031302d91fe703a19945a42e16620d4fb860604daf4"
+  license "BSD-4-Clause-UC"
+
+  depends_on "cmake" => :build
+  depends_on "make" => :build
```
Comment (YoshitakaMo) [inline Formula/reduce.rb:10]:
```suggestion
  depends_on "boost" => :build
  depends_on "cmake" => :build
```

`make` is not necessary. `boost => :build` may be required for this formula according to the cmake info.

File: Formula/reduce.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on "python"
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
```
Comment (YoshitakaMo) [inline Formula/reduce.rb:18]:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

This writing style is more recommended in Homebrew.

File: Formula/reduce.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  desc "Tool for adding and correcting hydrogens in PDB files"
+  homepage "https://github.com/rlabduke/reduce"
+  url "https://github.com/rlabduke/reduce/archive/refs/tags/v4.14.tar.gz"
+  version "4.14"
+  sha256 "62e61cce221fff76b5834031302d91fe703a19945a42e16620d4fb860604daf4"
+  license "BSD-4-Clause-UC"
+
+  depends_on "cmake" => :build
+  depends_on "make" => :build
+  depends_on "python"
```
Comment (YoshitakaMo) [inline Formula/reduce.rb:11]:
```suggestion
  depends_on "python@3.12"
```

python version is required.

File: Formula/reduce.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      system "cmake", "..", *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    output = shell_output("#{bin}/reduce -version 2>&1", 2)
+    assert_match "version 4.14", output
+  end
```
Comment (YoshitakaMo) [inline Formula/reduce.rb:24]:
```suggestion
  test do
    resource "homebrew-testdata" do
      url "https://files.rcsb.org/download/3QUG.pdb"
      sha256 "7b71128bedcd7ebdea42713942a30af590b3cf514726485f9aa27430c3999657"
    end

    output = shell_output(bin/"reduce -Version 2>&1", 2)
    assert_match "reduce.4.14.230914", output
    resource("homebrew-testdata").stage testpath
    system("#{bin}/reduce -NOFLIP -Quiet 3qug.pdb > 3qug_h.pdb")
    assert_match "add=1902, rem=0, adj=62", File.read("3qug_h.pdb")
  end
```

- The version info returned `reduce.4.14.230914` for `reduce -Version`. This result needs to be matched."
- It is encouraged that the test block describes the actual process and confirms that it works as expected.

**Review by YoshitakaMo — COMMENTED** [review]
Thank you for your contribution! I've described some suggestions in this PR to pass the CI. Please resubmit and check that the test is passed.

File: Formula/reduce.rb (hunk 2 above)
Comment (eunos-1128) [reply Formula/reduce.rb:10]: boost is not required for building C++ programs. 

It is needed only when building Python wrappers for the library, which the C++ program does not need.

**Comment by YoshitakaMo** [comment]
Unfortunately, the CI failed because a new rule was added to Homebrew just before (https://github.com/Homebrew/brew/commit/e6976ae3d1c79ce7cd5afa3f6739658eb2ae5992) . 
Please merge the latest commit of the `develop` branch and resubmit it. Thank you in advance.

**Comment by YoshitakaMo** [comment]
Merged! Thank you for your first PR! :beer:

**Comment by eunos-1128** [comment]
@YoshitakaMo 

Thank you for merging and your gracious help!

**Pull request merged**